cpp/dolfinx/fem/Function.h

```diff
@@ -400,15 +379,43 @@ class Function
     std::int32_t num_cells = cell_imap->size_local() + cell_imap->num_ghosts();
     std::vector<std::int32_t> cells(num_cells, 0);
     std::iota(cells.begin(), cells.end(), 0);
+    interpolate(e, cells, expr_mesh, cell_map);
+  }
 
-    if (cell_map.size() == 0)
-      interpolate(e, cells, expr_mesh, cell_map);
-    else
-      interpolate(e, cells, expr_mesh, std::span(cells.data(), cells.size()));
+  /// Interpolate a function defined on a non-matching mesh
+  /// @param[in] v The function to be interpolated
+  /// @param[in] padding Absolute padding for every bounding box on the mesh of
+  /// \p this to determine point ownership
+  /// @brief Evaluate the Function at points.
+  /// @note For repeated interpolation, it is adviced to precompute the
+  /// non-matching interpolation data
+  void
+  interpolate_nonmatching_meshes(const Function<value_type, geometry_type>& v,
+                                 T padding)
+  {
+    fem::interpolate_nonmatching_meshes(*this, v, padding);
+  }
+
+  /// Interpolate a function defined on a non-matching mesh
+  /// @param[in] v The function to be interpolated
+  /// @param cells THe cells in the mesh associated with @p u that will be
+  /// interpolated into
+  /// @param nmm_interpolation_data Data required for associating the
+  /// interpolation points of @p u with cells in @p v
+  /// @note For repeated interpolation, it is adviced to precompute the
+  /// non-matching interpolation data
+  void interpolate_nonmatching_meshes(
```

Comment:
This can be called interpolate - the compiler can resolve based on the arguments.